app/controllers/miq_request_controller.rb

```diff
@@ -39,7 +39,11 @@ def request_edit
       session[:checked_items] = provision_request.options[:src_ids]
       @refresh_partial = "reconfigure"
       @_params[:controller] = "vm"
-      reconfigurevms
+      if provision_request.kind_of?(VmCloudReconfigureRequest)
+        resizevms
```

Comment:
Why this? Is this really the place?

It doesn't sound right that reconfigure also does resize sometimes.

Is this a new feature? (The PR title does not seem to indicate so)

Reply:
The back end is "resize" (with request class VmCloudReconfigureRequest) for OpenStack and "reconfigure" (with request class VmReconfigureRequest) for other providers. The issue is that it was decided at some point in the past to expose these two completely different back-end actions as the same thing in the UI. On the model/back end side the actions are completely different, but from a user perspective, they have similar effects (in different providers). The result is that conditionals like this are necessary here.

This is not a new feature -- this PR fixes a bug -- the "edit request" action for administrators does not currently work for resize. A user can request a resize, and an admin can approve it as-is, but without this PR, the request cannot be edited.

Reply:
Aah, makes sense, thanks for the explanation! :)